baseplate/frameworks/queue_consumer/kombu.py

```diff
@@ -139,14 +139,17 @@ def handle(self, message: kombu.Message) -> None:
             amqp_exchange_name=message.delivery_info.get("exchange", ""),
             amqp_routing_key=message.delivery_info.get("routing_key", ""),
         )
-        AMQP_ACTIVE_MESSAGES.labels(**prometheus_labels._asdict()).inc()
 
         context = self.baseplate.make_context_object()
         try:
             # We place the call to ``baseplate.make_server_span`` inside the
             # try/except block because we still want Baseplate to see and
             # handle the error (publish it to error reporting)
-            with self.baseplate.make_server_span(context, self.name) as span:
+            with self.baseplate.make_server_span(
+                context, self.name
+            ) as span, AMQP_ACTIVE_MESSAGES.labels(
+                **prometheus_labels._asdict()
+            ).track_inprogress():
```

Comment:
This is a little difficult to read. Maybe something like this is better?

```suggestion
            with ( 
                self.baseplate.make_server_span(
                    context, 
                    self.name,
                ) as span,
                AMQP_ACTIVE_MESSAGES.labels(
                    **prometheus_labels._asdict(),
                ).track_inprogress()
            ):
```

Reply:
I thought the `with (...)` syntax was not introduced until later versions of python? But I can try

Reply:
This fails the linter. I'm pretty sure this syntax was only introduced in 3.10 too. That's also why we can't do things like `@h.labels(...).time()` because that's only supported in 3.10 as well.